### PR TITLE
Move functions into more focused source code modules

### PR DESCRIPTION
**Pull request by qkoziol**
Move functions into more focused source code modules, along with a small # of direct secondary effects.  No actual changes to the contents of any moved routines.

### CODE REVIEW

File: src/H5Cdbg.c [hunk 1]
```diff
@@ -1203,3 +1197,478 @@ H5C_verify_entry_type(H5C_t *cache_ptr, haddr_t addr, const H5C_class_t *expecte
...
+ *
+ *-------------------------------------------------------------------------
+ */
+void
+H5C_def_auto_resize_rpt_fcn(H5C_t *cache_ptr,
+#ifndef NDEBUG
+                            int32_t version,
+#else
+                            int32_t H5_ATTR_UNUSED version,
+#endif
```
Comment (jhendersonHDF) [inline src/H5Cdbg.c:1222]: I'm guessing you just moved this function from elsewhere, but it'd be nice to clean this up with H5_ATTR_NDEBUG_UNUSED instead if you could

Comment (qkoziol) [reply src/H5Cdbg.c:1222]: Yes - I didn't change any code for the functions themselves.

Agree about the change though, I'll put it on my TODO list and get to it in a future PR.

**Comment by qkoziol** [comment]
Anyone able to finish reviewing this, so I can get the next one up?

File: src/H5Cdbg.c [hunk 2]
```diff
@@ -713,7 +711,7 @@ void
 H5C_stats__reset(H5C_t *cache_ptr)
 #else /* NDEBUG */
 #if H5C_COLLECT_CACHE_STATS
-H5C_stats__reset(H5C_t *cache_ptr)
+H5C_stats__reset(H5C_t                            *cache_ptr)
```
Comment (derobins) [inline src/H5Cdbg.c:714]: ?

File: src/H5Cdbg.c (hunk 1 above)
Comment (derobins) [reply src/H5Cdbg.c:1222]: Should definitely fix, but also probably better to simply cut-n-paste around and make code changes in a subsequent PR

File: src/H5Centry.c [hunk 3]
```diff
@@ -0,0 +1,4363 @@
...
+ * distribution tree, or in https://www.hdfgroup.org/licenses.               *
+ * If you do not have access to either file, you may request a copy from     *
+ * help@hdfgroup.org.                                                        *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+/*-------------------------------------------------------------------------
+ *
+ * Created:     H5Centry.c
+ *              May 8 2023
+ *              Quincey Koziol
```
Comment (derobins) [inline src/H5Centry.c:17]: In the next PR, let's get rid of author fields. I've been removing them from everything I touch as they quickly become meaningless.

**Review by derobins — APPROVED** [review]

File: src/H5Cdbg.c (hunk 2 above)
Comment (qkoziol) [reply src/H5Cdbg.c:714]: Dunno - clang-format did that

File: src/H5Centry.c (hunk 3 above)
Comment (qkoziol) [reply src/H5Centry.c:17]: I'll add it to my TODO

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**